File: Problems/Parking Lot/Java/ticket/Ticket.java
package ticket;

import java.time.LocalDateTime;

import parkingSpot.ParkingSpot;
import java.util.UUID;

public class Ticket {
    private String ticketID;
    private LocalDateTime entryTime;
    private LocalDateTime exitTime;
    private ParkingSpot ps;

    public Ticket(ParkingSpot ps){
        this.ps = ps;
        this.ticketID = UUID.randomUUID().toString();
        this.entryTime = LocalDateTime.now(); 
    }

    public void setExitTime() {
        this.exitTime = LocalDateTime.now();
    }

    public String getTicketID() {
        return ticketID;
    }

    public ParkingSpot getParkingSpot() {
        return ps;
    }

    public String toString() {
        return "Ticket{" +
                "ticketID='" + ticketID + '\'' +
                ", entryTime=" + entryTime +
                ", exitTime=" + exitTime +
                ", parkingSpot=" + ps.getSpotID() +
                '}';
    }

}

File: Problems/Parking Lot/Golang/ticket/ticket.go
package ticket

import (
	parkingspot "parking-lot/parkingSpot"
	"time"

	"github.com/google/uuid"
)

type Ticket struct {
	TicketID    string
	ParkingSpot parkingspot.ParkingSpot
	EntryTime   time.Time
	ExitTime    time.Time
}

func NewTicket(parkingSpot parkingspot.ParkingSpot) *Ticket {

	return &Ticket{
		TicketID:    uuid.New().String(),
		ParkingSpot: parkingSpot,
		EntryTime:   time.Now(),
	}
}

func (t *Ticket) GetTicketID() string {
	return t.TicketID
}

func (t *Ticket) GetParkingSpot() parkingspot.ParkingSpot {
	return t.ParkingSpot
}

func (t *Ticket) GetEntryTime() time.Time {
	return t.EntryTime
}

func (t *Ticket) GetExitTime() time.Time {
	return t.ExitTime
}

func (t *Ticket) SetExitTime() {
	t.ExitTime = time.Now()
}
